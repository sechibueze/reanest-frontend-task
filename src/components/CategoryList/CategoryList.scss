@import "../../assets/theme/colors.scss";
.category-icons-card {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  border-radius: 0.5rem;
  background-color: $white;
  margin: 0.5rem;
  margin-top: 1rem;
  padding: 0.5rem;
}

.category-intro {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  border-bottom: 1px solid $light-gray;
  padding: 0.3rem 0rem;
}
.category-title {
  color: $light-gray;
}
.category-controls {
  color: $dark;
  font-weight: bold;
  .lt {
    padding-right: 0.5rem;
    font-weight: bold;
  }
}
.category-list {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: center;
  padding: 1rem 0rem;

  & li {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 0.5rem;
  }
}
.category-icon {
  margin: 0.75rem;
}
.category-text {
  font-size: 0.8rem;
  text-align: center;
  margin-top: 0.45rem;
}
