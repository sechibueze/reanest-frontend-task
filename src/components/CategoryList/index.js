import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import "./CategoryList.scss";
const CategoryList = () => {
  return (
    <div className="category-icons-card">
      <div className="category-intro">
        <span className="category-title"> Categoris</span>
        <span className="category-controls">
          <span className="lt">
            {" "}
            <FontAwesomeIcon icon={["fas", "angle-left"]} />{" "}
          </span>
          <span className="gt">
            <FontAwesomeIcon icon={["fas", "angle-right"]} />{" "}
          </span>
        </span>
      </div>
      <ul className="category-list">
        <li>
          <span className="category-icon">
            <FontAwesomeIcon
              icon={["far", "credit-card"]}
              size="2x"
              fixedWidth
            />{" "}
          </span>
          <span className="category-text"> Mobile &amp; internet data </span>
        </li>
        <li>
          <span className="category-icon">
            <FontAwesomeIcon
              icon={["fas", "broadcast-tower"]}
              size="2x"
              fixedWidth
            />{" "}
          </span>
          <span className="category-text"> Cable TV </span>
        </li>
        <li>
          <span className="category-icon">
            <FontAwesomeIcon icon={["far", "lightbulb"]} size="2x" fixedWidth />{" "}
          </span>
          <span className="category-text"> Power &amp; utility </span>
        </li>
      </ul>
    </div>
  );
};

export default CategoryList;
