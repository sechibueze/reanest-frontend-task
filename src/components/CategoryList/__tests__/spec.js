import { shallow } from "enzyme";
import CategoryList from "../index";

describe("<CategoryList /> renders ", () => {
  const component = shallow(<CategoryList />);
  const catList = component.find(".category-icons-card");
  it("<CategoryList /> matches snapshot", () => {
    expect(component).toMatchSnapshot();
    expect(catList.length).toEqual(1);
  });
});
