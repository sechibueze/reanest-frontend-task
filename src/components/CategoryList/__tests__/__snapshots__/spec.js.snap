// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryList /> renders  <CategoryList /> matches snapshot 1`] = `ShallowWrapper {}`;
